backend/substrapp/views/filters_utils.py

```diff
@@ -193,8 +193,11 @@ def filter_list(object_type, data, query_params):
                         hashes = [x['key'] for x in filtering_data]
 
                         if object_type == 'model':
-                            filtered_list = [x for x in filtered_list
-                                             if _get_model_tuple(x)['objective']['hash'] in hashes]
+                            filtered_list = [
+                                x for x in filtered_list
+                                if (x['testtuple'] and x['testtuple']['objective'] and
+                                    x['testtuple']['objective']['hash'] in hashes)
```

Comment:
Nice lazy evaluation but is it possible that a `testtuple` has no objective ?

Reply:
Objective can now only be attached to testtuple and no other tuple, so there's no implicit way of inferring the objective. It has to be there.